app/assets/javascripts/Components/one_time_annotations_table.jsx

```diff
@@ -0,0 +1,244 @@
+      dataType: 'json',
+    }).then(res => {
+      this.setState({
+        data: res,
+        loading: false
+      });
+    });
+  };
+
+  refreshAnnotations = () => {
+    $.ajax({
+      url: Routes.uncategorized_annotations_assignment_annotation_categories_path(this.props.assignment_id),
+      dataType: 'json',
+    }).then(res => {
+      this.setState({
+        data: res,
+        loading: false
+      })
+    });
+  };
+
+  editAnnotation = (annot_id, content) => {
+    let onSubmit = (formData) => {
+      return ($.ajax({
+        url: Routes.update_annotation_text_assignment_annotation_categories_path(this.props.assignment_id)+ '?id=' + annot_id,
+        data:{ ...formData },
+        method: "PUT",
+        remote: true,
+        dataType: "json",
+      }).always(() =>
+        {
+          this.setState({
+            annotationModal: INITIAL_ANNOTATION_MODAL_STATE
+          })
+          this.refreshAnnotations();
+        }
+      ));
+    };
+
+
+    let category_id = '';
+    let deduction = '';
+
+    this.setState({
+      annotationModal: {
+        ...this.state.annotationModal,
+        show: true,
+        content: content,
+        category_id,
+        isNew: false,
+        changeOneOption: category_id && !deduction,
+        onSubmit,
+        title: I18n.t("helpers.submit.create", {
+          model: I18n.t("activerecord.models.annotation.one"),
+        }),
+      },
+    });
+  };
+
+  removeAnnotation = (annot_id) => {
+    $.ajax({
+      url: Routes.destroy_annotation_text_assignment_annotation_categories_path(this.props.assignment_id)+ '?id=' + annot_id,
+      method: 'delete',
+      remote: true,
+      dataType: 'script'
+    }).then(() =>
+      {
+      this.refreshAnnotations()
+      });
+  };
+
+  columns = [
+    {
+      Header: I18n.t('activerecord.models.group.one'),
+      accessor: 'group_name',
+      id: 'group_name',
+      Cell: row => {
+        let remove_button = <a
+        href="#"
+        className="remove-icon"
+        title={I18n.t('delete')}
+        onClick={() => this.removeAnnotation(row.original.id, row.original.result_id)}
+      />
+        if (row.original.result_id) {
+          const path = Routes.edit_assignment_submission_result_path(
+            this.props.assignment_id,
+            row.original.submission_id,
+            row.original.result_id
+          );
+          return (
+            <div>
+            {remove_button}
+            <a className ="alignright" href={path}>{row.original.group_name}</a>
+          </div>
+          );
+        } else {
+          return (
+            <div>
+            {remove_button}
+            <span className="alignright">
+              {row.original.group_name}
+            </span>
+          </div>
+          );
+        }
+      },
+      filterMethod: (filter, row) => {
+        if (filter.value) {
+          // Check group name
+          if (row._original.group_name.includes(filter.value)) {
```

Comment:
Just `return` this expression (no need for if statement). Same for others below.